src/Razor/src/Microsoft.CodeAnalysis.Razor/EventHandlerTagHelperDescriptorProvider.cs

```diff
@@ -140,8 +140,8 @@ private List<TagHelperDescriptor> CreateEventHandlerTagHelpers(List<EventHandler
 
                     a.Name = attributeName;
 
-                    // Use a string here so that we get HTML context by default.
-                    a.TypeName = typeof(string).FullName;
+                    // We want event handler directive attributes to default to C# context.
+                    a.TypeName = typeof(object).FullName;
```

Comment:
👍

Reply:
Actually is it possible to use a more specific type here? Can we use `EventCallback<>` when we know the eventargs type, and `EventCallback` when we don't?

Reply:
@rynowak could you explain a little more? Can you give me an example of when we don't know the eventargs type?

Reply:
Hmmm maybe we always do. It's been a while since I wrote this codez.

Reply:
Okay. In that case, does it make sense to make the type name the following?
```cs
a.TypeName = $"Microsoft.AspNetCore.Components.EventCallback<{eventArgType}>";
```
I could be totally wrong but I thought we also support `Action<T>` instead of EventCallback. Will that still work?

Reply:
I think it's OK to *describe* the tag helpers in terms of event callback. Does it change the generated code?

Reply:
Okay it didn't change the generated code. I've made the change.